lib/factorial.rb

```diff
@@ -1,6 +1,15 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: On!
+# Space complexity: Constant
 def factorial(number)
-  raise NotImplementedError
-end
+  raise ArgumentError, "Bad Value" if number == nil
```

Comment:
If you want to also cover bad string integer input, consider:
`raise ArgumentError, "Bad Value" if number.is_a?(Integer) == false || number == nil `